src/Fixer/LanguageConstruct/NoUnsetOnPropertyFixer.php

```diff
@@ -71,142 +71,161 @@ protected function applyFix(\SplFileInfo $file, Tokens $tokens)
-    {
-        $index = 0;
-        $atLastUnset = false;
-        do {
-            $next = $tokens->getNextTokenOfKind($index, [',']);
-            if (null === $next) {
-                $atLastUnset = true;
-                $next = $tokens->count() - 1;
-                $nextTokenSet = $this->createTokensSubSet($tokens, $index, $next + 1);
-            } else {
-                $nextTokenSet = $this->createTokensSubSet($tokens, $index, $next);
-            }
-
-            if ($this->doTokensInvolveProperty($nextTokenSet)
-                && !$this->doTokensInvolveArrayAccess($nextTokenSet)
-            ) {
-                $replacement = $this->createReplacementIsNull($nextTokenSet);
-            } else {
-                $replacement = $this->createReplacementUnset($nextTokenSet);
-            }
-
-            $tokens->overrideRange($index, $next, $replacement);
-            $index = $tokens->getNextTokenOfKind($index, [';']) + 1;
-        } while (!$atLastUnset);
-    }
-
-    /**
-     * @param Tokens $filteredTokens
+     * @param int    $index
      *
-     * @return Token[]
+     * @return string[]
      */
-    private function createReplacementIsNull(Tokens $filteredTokens)
+    private function getUnsetsInfo(Tokens $tokens, $index)
     {
-        return array_merge(
-            $filteredTokens->toArray(),
-            [
-                new Token([T_WHITESPACE, ' ']),
-                new Token('='),
-                new Token([T_WHITESPACE, ' ']),
-                new Token([T_STRING, 'null']),
-                new Token(';'),
-            ]
-        );
+        $argumentsAnalyzer = new ArgumentsAnalyzer();
+
+        $unsetStart = $tokens->getNextTokenOfKind($index, ['(']);
+        $unsetEnd = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $unsetStart);
+        $isFirst = true;
+
+        $unsets = [];
+        foreach ($argumentsAnalyzer->getArguments($tokens, $unsetStart, $unsetEnd) as $startIndex => $endIndex) {
+            $startIndex = $tokens->getNonWhitespaceSibling($startIndex - 1, 1);
+            $endIndex = $tokens->getNonWhitespaceSibling($endIndex + 1, -1);
+            $unsets[] = [
+                'startIndex' => $startIndex,
+                'endIndex' => $endIndex,
+                'isToTransform' => $this->isProperty($tokens, $startIndex, $endIndex),
+                'isFirst' => $isFirst,
+            ];
+            $isFirst = false;
+        }
+
+        return $unsets;
     }
 
     /**
-     * @param Tokens $filteredTokens
+     * @param Tokens $tokens
+     * @param int    $index
+     * @param int    $endIndex
      *
-     * @return Token[]
+     * @return bool
      */
-    private function createReplacementUnset(Tokens $filteredTokens)
+    private function isProperty(Tokens $tokens, $index, $endIndex)
     {
-        if ($filteredTokens[0]->isWhitespace()) {
-            $filteredTokens->clearAt(0);
+        while ($tokens[$index]->isWhitespace() || $tokens[$index]->isComment()) {
```

Comment:
Can we use `getNextMeaningfulToken` here instead?

Reply:
We can even remove it with updating parent call.